File: active_subspaces/Matlab_Version/Python_Testing/Data_Set/test_script_SDR.py
import utils_data
import numpy as np
import active_subspaces.sdr
import matplotlib.pyplot as plt

# Set quantity of interest and input labels
QofI = 'QoI'
in_labels = ['rho', 'mu', 'dpdz', 'c_p', 'k', 'Pr_t']

# Set the number of parameter (m) and active subspace dimension (n)
m, n = 6, 3
# Load random sampling data
data = np.loadtxt('params_and_results.dat')
X_phys, f, df_phys = data[:, :m], data[:, m].reshape((data.shape[0], 1)), data[:, m+1:]
M = f.shape[0]

# Scale data so parameters are in [-1, 1]
X_norm, df_norm = utils_data.physical_to_normalized(X=X_phys, df=df_phys)

# Test linear gradient check
w = active_subspaces.sdr.linear_gradient_check(X_norm, f)

# Test quadratic model check
gamma = (1.0/3.0)*np.ones((1, m))
e, W = active_subspaces.sdr.quadratic_model_check(X_norm, f, gamma)

# Plot the active subspace info
active_subspaces.utils.plotters.eigenvalues(e, out_label=QofI)
active_subspaces.utils.plotters.eigenvectors(W[:, :n], in_labels=in_labels, out_label=QofI)
active_subspaces.utils.plotters.sufficient_summary(np.dot(X_norm, W[:, :2]), f, out_label=QofI)

File: matlab/Demos/test_script_SDR.m
clear variables
close all
clc

addpath 'Subspaces' 'SDR' 'Plotters'

%% Test functions - Choose 1

% Linear function
m = 5; n = 2;
M = 100;
X = 2*rand(M, m) - 1;
[f, df] = test_function_1(X);
df = df./repmat(sqrt(sum(df.^2, 2)), 1, m);

% Data Set
% [X, f, df] = test_function_2();
% [M, m] = size(X);
% n = 2;
% df = df./repmat(sqrt(sum(df.^2, 2)), 1, m);

%% Test linear_gradient_check.m

w = linear_gradient_check(X, f);

%% Test quadratic_model_check.m

gamma = ones(m, 1)/3;
[e, W] = quadratic_model_check(X, f, gamma);

eigenvalues(e)
eigenvectors(W(:, 1:2))
sufficient_summary(X*W(:, 1:2), f)